controller/monitor/monitor.go

```diff
@@ -12,9 +12,9 @@ import (
 type Monitor interface {
 	Start()
 	Close()
-
-	SyncCollectedData() error
-	GetCollectedData() (interface{}, error)
+	Update(map[string]interface{}) error
+	RunTask(interface{}) error
```

Comment:
Of course, for any downstream structure, it can override all methods for it's behavior, but because we have baseMonitor, can think about how to make reusable code to the parent class.

Reply:
The purpose of exporting `Run` is to do the unit test, but I also think it not good here. I think we can make it private when refractorig `backup` and `engine` monitors. Then, we can make the methods more generalized.

`GetData` and `Update` will by called by controllers for retrieving the information from monitors or upadte the monitors' configurations.

Reply:
Renamed the methods. Thank you.